### ISSUE TITLE
Cannot withdraw notifications from previous session

### ISSUE THREAD

**Issue by latenitefilms** [Hammerspoon/hammerspoon#2241]
If I create a bunch of `hs.notify` notifications then restart Hammerspoon, I expect the below code should work to remove these notifications after restarting:

```lua
local deliveredNotifications = hs.notify.deliveredNotifications()
for _, notification in pairs(deliveredNotifications) do
    print(string.format("Withdrawing: %s", notification))
    notification:withdraw()
end
```

...however, sadly it doesn't.

I can see "hs.notify: creating new userdata for orphaned notification" in Console.app for each notification that's already on the screen.

`hs.notify.withdrawAll()` works as expected.

Any ideas @asmagill?

**Comment by cmsj** [comment]
Slightly separately from this specific issue, and while it wouldn't help with crashes, I'm wondering if we should unconditionally withdraw all notifications when HS exits - or perhaps have a default shutdown callback that does it, so people can override the behaviour if they like littering their screen with orphaned notifications.

**Comment by latenitefilms** [comment]
Ummm, I'm not sure about that. I think it's better to leave that up to the end-user to solve. In our case, I want Hammerspoon to "reconnect" to those existing notifications when restarting, but also have a way to detect when they've been closed when Hammerspoon wasn't running.

AFAIK this used to work correctly - so I think it's maybe a Mojave-related issue?

**Comment by asmagill** [comment]
Will be addressed with #2691

**Comment by latenitefilms** [comment]
Untested, but assuming this is fixed in #2691 ?
